docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -0,0 +1,136 @@
+
+### As a user, I want to migrate from Kuma non-SPIFFE complient certificates to SPIRE
+
+We should allow users to migrate from Kuma issued certificates, which are not SPIFFE-compliant, to using SPIRE. This may require an intermediate step to first migrate to SPIFFE-compliant certificates, but the details will be specified later.
+
+## Future User Stories (things that are going to be implemented in future releases)
+
+### As a user, I want to have a SPIFFE-compliant SAN in a non-Kubernetes environment
+
+It would be beneficial to support connectivity between different environments. Unfortunately, our services currently lack proper identity on Universal, which makes this challenging.
+
+### As a user, I want to federate different trust domains using the Federation API exposed by the control plane
+
+It should be possible to federate different trust domains without creating additional resources, only by using the federation endpoint exposed by the control plane. This would be a valuable feature, enabling interoperability between different SPIFFE-compliant systems.
+
+### As a user, I want private key to not be transfered over network
+
+The private key should not be sent from the control plane to the dataplane. This can be addressed by redesigning our secret flow: instead of pushing certificates from the control plane, the dataplane should send a Certificate Signing Request (CSR) to the control plane, which then signs the certificate. In this setup, the private key remains local and never leaves the dataplane environment. This change may require significant work and might not be feasible in the current release, but it should be targeted for a future release.
+
+### As a user, I want to use an intermediate CA instead of a root CA on the zone
+
+When a user provides a root CA on the global control plane, we could generate a separate intermediate CA for each connecting zone. This would offer a stronger security model, ensuring that if a certificate is compromised in one zone, it does not impact other zones.
+
+## Out of scope
+
+### As a user, I want to specify certificates for a specific outbound
+
+You should use MeshExternalService in this case to configure the outbound connection and provide the required certificates.
+
+### As a user, I want my dataplanes to communicate with Istio services without going through a gateway
+
+In this scenario, you should also use MeshExternalService and include the necessary certificates in the configuration to enable direct mTLS communication.
+
+### As a user, I want to use two different sources of CA (Kuma and SPIRE) at the same time
+
+This is a complex use case and can make the configuration error prone and difficult to manage.
+We want to add some limitation to avoid issues:
+* A dataplane's identity must come from a single source.
+* For trust domains, identities must be issued by one system only — either SPIRE or Kuma, never both for the same dataplane.
```

Comment:
We're mixing things here I think.
I think we're getting the info (identity and trust) either from Spire or Kuma never both.
However, it's possible to add the trust coming from Spire inside Kuma and to implement SPIRE federation api in Kuma to retrieve Trust.

Reply:
Rephrased